packages/polkadart_scale_codec/test/test_types/signed_int_test.dart

```diff
@@ -0,0 +1,182 @@
+import 'package:polkadart_scale_codec/src/core/core.dart';
+import 'package:polkadart_scale_codec/src/util/utils.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('encode to hex signed int tests', () {
+    final typesRegistry = <String, dynamic>{
+      'Codec': {
+        'a': 'i8',
+        'b': 'i16',
+        'c': 'i32',
+      },
+    };
+    // Creates the registry for parsing the types and selecting particular schema.
+    final registry = OldTypeRegistry(types: typesRegistry);
+
+    // specifyng which schema type to use
+    registry.select('Codec');
+
+    // fetching the parsed types from `Json` to `Type`
+    final types = registry.getTypes();
+
+    // Initializing Scale-Codec object
+    final codec = Codec(types);
+
+    group('fixed-length 8bit', () {
+      test('Should encode and return correct value ', () {
```

Comment:
Please improve titles. Same for other tests
```suggestion
      test('Given a positive integer when it is within range it should be encoded', () {
```